src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx

```diff
@@ -0,0 +1,247 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2023 Gemeente Amsterdam
+
+import { act, fireEvent, render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import * as reactRedux from 'react-redux'
+import departmentsFixture from 'utils/__tests__/fixtures/departments.json'
+
+import { Filter } from './Filter'
+
+const mockCallback = jest.fn()
+
+window.HTMLElement.prototype.scrollIntoView = jest.fn()
+
+const departments = {
+  ...departmentsFixture,
+  count: departmentsFixture.count,
+  list: departmentsFixture.results,
+}
+
+describe('FilterComponent', () => {
+  beforeEach(() => {
+    jest.spyOn(reactRedux, 'useSelector').mockReturnValue(departments)
+  })
+
+  it('should render properly and select the first option', () => {
+    render(<Filter callback={mockCallback} />)
+
+    userEvent.click(screen.getAllByRole('listbox')[0])
+    expect(
+      screen.getByRole('listbox', { name: 'Actie Service Centr...' })
+    ).toBeInTheDocument()
+
+    expect(
+      screen.queryByRole('listbox', {
+        name: 'Amsterdamse Bos',
+      })
+    ).not.toBeInTheDocument()
+  })
+
+  it('should clear filterActiveName when clicking outside Filter', () => {
+    render(
+      <>
+        <button>outside filter</button>
+        <Filter callback={mockCallback} />
+      </>
+    )
+
+    userEvent.click(
+      screen.getByRole('listbox', {
+        name: 'Actie Service Centr...',
+      })
+    )
+
+    userEvent.click(
+      screen.getByRole('button', {
+        name: 'outside filter',
+      })
+    )
+
+    expect(
+      screen.queryByRole('option', { name: 'Amsterdamse Bos' })
+    ).not.toBeInTheDocument()
+  })
+
+  it('should select a department and thus change selectable categories', async () => {
+    render(<Filter callback={mockCallback} />)
+
+    expect(
+      screen.getByRole('listbox', {
+        name: 'Actie Service Centr...',
+      })
+    ).toBeInTheDocument()
+
+    userEvent.click(
+      screen.getByRole('listbox', {
+        name: 'Categorie',
+      })
+    )
+
+    expect(
+      screen.queryByRole('option', { name: 'Bedrijfsafval' })
+    ).toBeInTheDocument()
+
+    userEvent.click(
+      screen.getByRole('listbox', {
+        name: 'Actie Service Centr...',
+      })
+    )
+
+    userEvent.click(
+      screen.getByRole('option', {
+        name: 'Politie',
+      })
+    )
+
+    expect(
+      screen.queryByRole('listbox', { name: 'Bedrijfsafval' })
+    ).not.toBeInTheDocument()
+
+    expect(
+      screen.queryByRole('listbox', { name: 'Categorie' })
+    ).toBeInTheDocument()
+  })
+
+  it('should select a department and reset form by using keyboard', () => {
```

Comment:
This does not work as expected. When you press enter on "Categories" the dropdown appears, but then on Tab you go to the next item in the menu instead of going through the options of "Categories"

Reply:
yeah I should focus

Reply:
This does no work as expected yet. See comments in the main section.